### PR TITLE
Add Airtrain + Llama Index cookbook

### PR DESCRIPTION
**Pull request by augray**
# Description

This PR adds a cookbook which describes how to use Airtrain with Llama Index. It introduces no new dependencies for any Llama Index packages, but the notebook for the cookbook depends on the Airtrain python package.

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
(^ notebook requires user-specific API keys, but otherwise can be executed as-is end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
(^ in the sense that the notebook itself is a test)
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/docs/examples/cookbooks/airtrain.ipynb [hunk 1]
```diff
@@ -0,0 +1,507 @@
+{
```
Comment (nerdai) [inline docs/docs/examples/cookbooks/airtrain.ipynb:1]: <p>Can you kindly do a string search and replace in this notebook to change "Llama Index" (with space) to "LlamaIndex" (without space).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/16332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (nerdai) [inline docs/docs/examples/cookbooks/airtrain.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.            if not isinstance(ev, CompletedDocumentRetrievalEvent):
</pre><p>It's actually not necessary to check the type of event. Under the hood, inspection is being done on the step's to determine its accepted events.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/16332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (nerdai) [inline docs/docs/examples/cookbooks/airtrain.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #22.            return AirtrainDocumentDatasetEvent(metadata=dataset_meta)
</pre><p>you can probably just return <code>StopEvent(result=metadata)</code> here and omit the <code>complete_workflow</code> step.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/16332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

**Review by nerdai — CHANGES_REQUESTED** [review]
Looks good @augray, though I have left a few minor comments.

File: docs/docs/examples/cookbooks/airtrain.ipynb (hunk 1 above)
Comment (augray) [reply docs/docs/examples/cookbooks/airtrain.ipynb:1]: <p>done!</p>

Comment (augray) [reply docs/docs/examples/cookbooks/airtrain.ipynb:1]: <p>removed this</p><p> </p>

Comment (augray) [reply docs/docs/examples/cookbooks/airtrain.ipynb:1]: <p>done!</p>

**Comment by augray** [comment]
Thanks for the review @nerdai ; I applied your suggested changes

**Review by nerdai — APPROVED** [review]
Looks great! Thanks @augray

**Pull request merged**